jena-text/src/main/java/org/apache/jena/query/text/TextIndexLucene.java

```diff
@@ -112,6 +112,7 @@
     private final FieldType        ftTextNotStored ; // used for lang derived fields
     private final FieldType        ftTextStoredNoIndex ; // used for lang derived fields
     private final boolean          isMultilingual ;
+    private final int              maxBasicQueries ;
```

Comment:
This should specify a value.

When used via `TextIndexLuceneAssembler`, it will be set to a default of 1024. jena-text may be used from code , no assembler, and then the default is zero from this line.

It is always passed in `BasicQueryFactory(this.maxBasicQueries)` by `TextLuceneIndex` so the other choice is set to -1 (an "unset" marker) and call `BasicQueryFactory()` when not set so that it tracks the Lucene choice for the default which may evolve over time.

Reply:
I think L#136, below, ensures a value for ```maxBasicQueries```. Perhaps L#30 above in ```TextIndexConfig.java``` should be set to 1024 since that's the default in ```org.apache.lucene.queryparser.surround.query. BasicQueryFactory ```

Reply:
Yours is a better explanation. I hadn't made the extra step to `TextIndexConfig`.

It gets a default from `TextIndexConfig` which itself may not have a default other than Java's 0. Only a call of `setMaxBasicQueries` sets it and that only comes from the assembler

Conclusion - it needs an explicit default (both places) so we can understand it ! :-)

Plan: Let this sit for at least 24 hours then merge it and make the changes. This has been here so long that I'm happy to do the changes directly. We've kept @DamienFontaine's PR waiting.

OK @xristy ? If that is fine with you, would you approve please?

Reply:
Fine w/ me. I would vote for 1024 rather than -1 for the default. I approved changes on my request for changes.